zyte_common_items/pipelines.py

```diff
@@ -53,3 +55,67 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropLowProbabilityItemPipeline:
+    """This pipeline drops an item if its probability, defined in the settings,
+    is less than the specified threshold.
+
+    There are two ways that thresholds can be defined per item type using
+    the `ITEM_PROBABILITY_THRESHOLDS` setting:
+
+    With paths to Item classes:
+        ITEM_PROBABILITY_THRESHOLDS = {
+            "zyte_common_items.items.Article": 0.1,
+            "zyte_common_items.items.Product": 0.2
+        }
+
+    With Item classes:
+        from zyte_common_items import Article, Product
+        ITEM_PROBABILITY_THRESHOLDS = {
+            Article: 0.1,
+            Product: 0.2
+        }
+    """
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def init_thresholds(self, spider):
+        from scrapy.utils.misc import load_object
+
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = load_object(item) if isinstance(item, str) else item
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
+        return self.thresholds.get(type(item), DEFAULT_ITEM_PROBABILITY_THRESHOLD)
+
+    def get_item_name(self, item):
+        item_name = item.__class__.__name__
+        return item_name[0].lower() + item_name[1:]
+
+    def process_item(self, item, spider):
+        from scrapy.exceptions import DropItem
+
+        threshold = self.get_threshold(item, spider)
+
+        self.stats.inc_value("item/crawl/total", spider=spider)
+        item_proba = item.get_probability()
+        if item_proba is None or item_proba >= threshold:
+            return item
+
+        self.stats.inc_value(
+            f"drop_item/{self.get_item_name(item)}/low_probability", spider=spider
+        )
+        raise DropItem(
+            f"The item: {item!r} is dropped as the probability ({item_proba}) "
+            f"is below the threshold ({threshold})"
```

Comment:
```suggestion
            f"This item is dropped since the probability ({item_proba}) "
            f"is below the threshold ({threshold}):\n{item!r}"
```
Minor: just so it's easier to read the logs as the sentence comes first containing the proba and threshold values before the actual item value which can be quite long. Otherwise, the reader might need to scroll down at the end to see such values.